src/main/java/com/ericsson/ei/subscription/SubscriptionValidator.java

```diff
@@ -70,29 +76,32 @@ public static void validateSubscription(Subscription subscription) throws Subscr
     }
 
     /**
-     * Validation of subscriptionName parameter Throws SubscriptionValidationException if validation of
-     * the parameter fails due to wrong format of parameter.
+     * Validation of subscriptionName parameter Throws
+     * SubscriptionValidationException if validation of the parameter fails due to
+     * wrong format of parameter.
      *
      * @param subscriptionName
      */
     private static void validateSubscriptionName(String subscriptionName) throws SubscriptionValidationException {
-        // When this regExp need to be changed then remember to change the one in the
-        // back-end (invalidSubscriptionNameRegex in subscription.js), which do the same
-        // invalid subscription name check. The two
-        // regEx always need to be the same for ensuring the same check.
-        // /(\W)/ Is a regEx that matches anything that is not [A-Z,a-z,0-8] and _.
-        String invalidSubscriptionNameRegex = "(\\W)";
+        // Here the regEx used is :/(\W)/ that matches anything that is not
+        // [A-Z,a-z,0-8] and _.
+        String invalidSubscriptionNameRegex = null;
+        invalidSubscriptionNameRegex = RegExProvider.SUBSCRIPTION_NAME;
```

Comment:
According to the comment above the regex does not match a subscription name, but everything else right? Then I find is misleading that the regex name for it says "SUBSCRIPTION_NAME" implying that the regex contains the opposite ... 

Either change the name of the constant or, what I prefer, change the regex to contain only allowed values and then update this check.

Reply:
Well, this specific regex was used (I think) to spill out the wrong characters, especially in the front-end subscription form, to inform user. The name, I agree it doesnt give the clear picture, is used to match the one used in the subscription object. So if other agree to use different name then I will change it.